DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/compute_units/src/pipe_array.hpp

```diff
@@ -0,0 +1,43 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <utility>
+
+// Header locations and some DPC++ extensions changed between beta09 and beta10
+// Temporarily modify the code sample to accept either version
+#define BETA09 20200827
+#if __SYCL_COMPILER_VERSION <= BETA09
+  #include <CL/sycl/intel/fpga_extensions.hpp>
+  namespace INTEL = sycl::intel;  // Namespace alias for backward compatibility
+#else
+  #include <CL/sycl/INTEL/fpga_extensions.hpp>
+#endif
+
+
+#include "pipe_array_internal.hpp"
+
+template <class Id, typename BaseTy, size_t depth, size_t... dims>
+struct PipeArray {
+  PipeArray() = delete;
```

Comment:
Objects of type PipeArray should never be created. This line makes it impossible to accidentally construct PipeArrays, by disabling the default constructor for PipeArray. Since we didn't add any other constructors, it won't be possible to construct PipeArray at all.

Reply:
Thanks. But what I mean is, explain this in a comment. :-) Keep it for your next improvement. This is a **tutorial**, so anyone interested by FPGA (usually not super C++-savvy) should be able to understand this code.

Reply:
Is this the definition of a singleton class?

Reply:
No. Just that this class is only used as a meta-function for meta-programming and its instantiation has no usage.
The problem here is that since nothing is explained, a user trying to create such an object could not, but not knowing why, since there is no `static_assert` or comment to clarify the reason.
On the other hand, creating such an object has no effect, so at the end this is a lot of work to just add some smoke curtain... Perhaps just removing this line is good? :-)

Reply:
I agree this is confusing. Having a class called `PipeArray` but you cannot make an object seems, at best, counterintuitive.  I would think I could make as many of these as I want... Maybe we can figure something cleaner out later @jessicadavies-intel